### PR TITLE
Improve the Release RST script to support things other than SunPy Core

### PR DESCRIPTION
**Pull request by Cadair**
This adds much more robust argument parsing and other niceties to the release notes script.

### CODE REVIEW

File: tools/generate_releaserst.xsh [hunk 1]
```diff
@@ -167,33 +172,28 @@ for i, line in enumerate(lines):
...
-print(f"This release of SunPy contains {ncommits} commits in {prcnt['merged']} merged pull requests closing {icnt['closed']} issues from {npeople} people, {nnew} of which are first time contributors to SunPy.")
+pretty_project_name = args["--pretty-project-name"] if args["--pretty-project-name"] else args["--project-name"]
+print(f"This release of {pretty_project_name} contains {ncommits} commits in {prcnt['merged']} merged pull requests closing {icnt['closed']} issues from {npeople} people, {nnew} of which are first time contributors to {pretty_project_name}.")
 print()
 print("The people who have contributed to the code for this release are:")
 print()
 print(output)
 print()
-print("Where a * indicates their first contribution to SunPy.")
+print(f"Where a * indicates their first contribution to {pretty_project_name}.")
```
Comment (bsipocz) [inline tools/generate_releaserst.xsh]: can pretty_project_name default to project_name rather to `Sunpy`?

**Comment by bsipocz** [comment]
@Cadair - would you be interested in pushing this back to astropy-procedures, too?

**Comment by nabobalis** [comment]
Do we even need this in core sunpy repo?

**Comment by bsipocz** [comment]
> Do we even need this in core sunpy repo?

given it's super generic now, if you prefer it can live in astropy-procedures. Or maybe in openastronomy.

**Comment by Cadair** [comment]
I guess we need an OpenAstronmy/procedures repo :grinning:

**Comment by bsipocz** [comment]
> I guess we need an OpenAstronmy/procedures repo

that would makes sense I suppose

**Comment by Cadair** [comment]
fwiw, I will swing back to this when I get a chance, but I didn't think it was worth having the PR hanging open.

**Pull request merged**